compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -161,15 +198,47 @@ instance Eq AnyContractKey where
 #endif
 
 #ifdef DAML_TYPE_REP
+
+class HasTemplateTypeRep t where
+  -- | Generate a unique textual representation of the template Id.
+  _templateTypeRep : proxy t -> TypeRep
+
 -- | Generate a unique textual representation of the template Id.
-templateTypeRep : forall t. Template t => TemplateTypeRep
-templateTypeRep = _templateTypeRep ([] : [t])
+templateTypeRep : forall t. HasTemplateTypeRep t => TemplateTypeRep
+templateTypeRep = TemplateTypeRep (_templateTypeRep ([] : [t]))
 
 instance Eq TemplateTypeRep where
   TemplateTypeRep a == TemplateTypeRep b = a == b
 
 #ifdef DAML_ANY_TYPE
 
+class HasToAnyTemplate t where
```

Comment:
I think you need to move the definition of those classes out of the `ifdef`. The template desugaring does not depend on the LF version so we will always generate instances for this.

Reply:
Ah, good point! I also removed the CPP conditions on the respective constraints in `Template`, `Choice`, and `TemplateKey`, since otherwise I think it might be a bit of a *gotcha* without good reason.